pkg/plugins/neutron.go

```diff
@@ -129,7 +131,24 @@ func init() {
 
 //Init implements the core.QuotaPlugin interface.
 func (p *neutronPlugin) Init(provider *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) error {
-	_, err := openstack.NewLoadBalancerV2(provider, eo)
+	client, err := openstack.NewNetworkV2(provider, eo)
+	if err != nil {
+		return err
+	}
+
+	// LBaaSv2 supported?
+	r := extensions.Get(client, "lbaasv2")
+	switch r.Result.Err.(type) {
+	case gophercloud.ErrDefault404:
+		p.hasLBaaS = false
+	case nil:
+		p.hasLBaaS = true
+	default:
+		return r.Result.Err
```

Comment:
A more descriptive error message would be nice.

```suggestion
		return fmt.Errorf("cannot check for lbaasv2 support in Neutron: %s", r.Result.Err.Error())
```